internal/testrunner/runners/system/runner.go

```diff
@@ -1150,9 +1159,15 @@ func (r *runner) previewTransform(transformId string) ([]common.MapStr, error) {
 
 func deleteDataStreamDocs(api *elasticsearch.API, dataStream string) error {
 	body := strings.NewReader(`{ "query": { "match_all": {} } }`)
-	_, err := api.DeleteByQuery([]string{dataStream}, body)
+	resp, err := api.DeleteByQuery([]string{dataStream}, body)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to delete data stream docs: %w", err)
+	}
+	defer resp.Body.Close()
+
+	// Not found error is fine here, this means that data was already not there.
+	if resp.IsError() && resp.StatusCode != http.StatusNotFound {
+		return fmt.Errorf("failed to delete data stream docs for data stream %s: %s", dataStream, resp.String())
 	}
```

Comment:
Should it be managed the other cases of `resp.IsError()` ?

Reply:
503 needed to be handled, because of https://github.com/elastic/elasticsearch/issues/65846. Other errors should be just returned.

Reply:
Ah no, 503 needed to be handled when searching in `getDocs`. Here I think that we don't need to handle other cases.